FlagsmithClient/src/test/java/com/flagsmith/RealTimeUpdatesIntegrationTests.kt

```diff
@@ -0,0 +1,216 @@
+            )
+        ).thenReturn(
+            mockSharedPreferences
+        )
+        Mockito.`when`(mockApplicationContext.cacheDir).thenReturn(File("cache"))
+
+        Mockito.`when`(mockContextResources.getString(ArgumentMatchers.anyInt())).thenReturn("mocked string")
+        Mockito.`when`(mockContextResources.getStringArray(ArgumentMatchers.anyInt())).thenReturn(
+            arrayOf(
+                "mocked string 1",
+                "mocked string 2"
+            )
+        )
+        Mockito.`when`(mockContextResources.getColor(ArgumentMatchers.anyInt())).thenReturn(Color.BLACK)
+        Mockito.`when`(mockContextResources.getBoolean(ArgumentMatchers.anyInt())).thenReturn(false)
+        Mockito.`when`(mockContextResources.getDimension(ArgumentMatchers.anyInt())).thenReturn(100f)
+        Mockito.`when`(mockContextResources.getIntArray(ArgumentMatchers.anyInt()))
+            .thenReturn(intArrayOf(1, 2, 3))
+    }
+
+    @Test
+    fun testEnvironmentVariablesArentEmpty() {
+        Assert.assertTrue(environmentKey.isNotEmpty())
+        Assert.assertTrue(apiToken.isNotEmpty())
+        Assert.assertTrue(featureId.isNotEmpty())
+        Assert.assertTrue(featureStateId.isNotEmpty())
+    }
+
+    /// Update after 5 secs, should be done in 60 seconds or fail
+    @Test(timeout = 60000)
+    fun testGettingFlagsWithRealtimeUpdatesAfterPuttingNewValue() = runBlocking {
+        // Get the current value
+        val currentFlagValueString =
+            flagsmith.getValueForFeatureSync(featureId).getOrThrow() as String?
+        println("Type of currentFlagValueDouble: ${currentFlagValueString?.javaClass?.name}")
+        Assert.assertNotNull(currentFlagValueString)
+        val currentFlagValue: String = currentFlagValueString!!
+
+        // After 5 seconds try to update the value using the retrofit service
+        CoroutineScope(Dispatchers.IO).launch {
+            // Wait 5 seconds before updating the value
+            delay(5000)
+
+            val response = retrofitService
+                .setFeatureStates(authToken, featureStateId, environmentKey!!, FeatureStatePutBody(true, "new-value"))
+                .execute()
+            if (!response.isSuccessful) println("ERROR response: $response")
+            println("Response: $response")
+            Assert.assertTrue(response.isSuccessful)
+        }
+
+        var newUpdatedFeatureValue: String? = ""
+        do {
+            newUpdatedFeatureValue = flagsmith.flagUpdateFlow.value
+                .find { flag -> flag.feature.name == featureId }?.featureStateValue as String? ?: ""
+            delay(300L) // Delay a little while to give the CPU some time back
+        } while (newUpdatedFeatureValue.isNullOrEmpty() || newUpdatedFeatureValue == currentFlagValueString)
+
+        Assert.assertEquals("new-value", newUpdatedFeatureValue)
+    }
+
+    // Update after 65 secs to ensure we've done a reconnect, should be done in 100 seconds or fail
```

Comment:
I assume it's the network or some other bit of kit that times us out. I had this at 60 to cover various bits of kit. Have changed to 35.

Reply:
I am still seeing 65 secs in the comment here at least?

Reply:
@matthewelwell it's because the test was failing every so often with the reduced value, so I changed this back. I assume this is because it's taking a little while for the infrastructure to tell us about the new value via the socket.